feat: Checkout Page

bzgaspar opened the pull request:
Checkout Form. Add input validation. Add summary. Add Thank you modal. After clicking continue, remove all cart Items. 
Responsive from mobile to desktop.
-improvements-
Add button to cart items so if click the product can be shown. 
The top nav will close if categories are clicked.


[Preview](https://audiophile-e-commerce-website-git-feature-checkoutpage-itsmenek.vercel.app?_vercel_share=e0rFNpdiJ4NbHxdP8Tk6Z19oizWdKxxK)

bzgaspar committed:
feat: Checkout Page

Checkout Form. Add input validation. Add summary. Add Thankyou modal. After clicking continue, remove all cart Items. Responsive from mobile to desktop.

eremocal commented on src/components/CheckoutForm/CheckoutForm.tsx:
```diff
@@ -0,0 +1,202 @@
+            />
+          </div>
+        </div>
+        {choice === 'e-Money' ? (
```
please create assign it to a variable or create an enum same with the others

eremocal commented on src/components/CheckoutForm/Summary/Summary.tsx:
```diff
@@ -0,0 +1,86 @@
+const SHIPPING = 50
+export default function Summary() {
+  const { cartItems, totalCartPrice, formValid } = useCartContext()
+  function getVatIncluded(total: number) {
```
please move this to utils file

eremocal commented on src/components/CheckoutForm/Summary/Summary.tsx:
```diff
@@ -0,0 +1,86 @@
+  function getVatIncluded(total: number) {
+    return total * 0.2
+  }
+  function renderGrandTotal(total: number) {
```
same with this

eremocal reviewed (commented):
![image](https://github.com/itsmeNEK/audiophile-e-commerce-website/assets/61583701/e3c571aa-484b-44dc-bd34-56723624fef1)
if there are no products in the cart are we going to show the fields?

bzgaspar committed:
refactor:

Move getVatIncluded and getGrandTotal to utils. Create variable for payment options

bzgaspar committed:
Assign e-Money and Cash on Delivery to enum

bzgaspar committed:
Refactor handleChoiceChange to set choice
